02-intermediate-ruby/exercises/more-testing-practice.md

```diff
@@ -0,0 +1,48 @@
+# Test Writing Exercise
+
+In this exercise you will look at a problem description and write a set of tests to evaluate a solution.  Then you will review the tests of another classmate and make suggestions for improvement.  Lastly you will write solution code to satisfy your tests.
+
+## Exercise Goal
+
+The goal of this set of exercises is to:
+
+*   Practice writing tests
+*   Review the tests written by another and look for:
+  -Testing all edge-cases
+  -Ways to DRY-up tests
+
+## Flatten Array Problem Description
+
+This method should take a nested list and return a single flattened list with all values except nil/null.
+
+The challenge is to write a function that accepts an arbitrarily-deep nested list-like structure and returns a flattened structure without any nil/null values.
+
+For Example:
+
+Ex 1:
+input: `[1,[2,3,null,4],[null],5]`
```

Comment:
I think we should use `nil` if we're planning on giving this to them within the context of their time in Ruby. Then we could avoid any confusion with the nil/null section above.